src/osd/ReplicatedPG.cc

```diff
@@ -10180,13 +10179,14 @@ int ReplicatedPG::recover_primary(int max, ThreadPool::TPHandle &handle)
 
 	      ++active_pushes;
 
-	      osd->store->queue_transaction(osr.get(), t,
+	      osd->store->queue_transaction(osr.get(), std::move(*t),
 					    new C_OSD_AppliedRecoveredObject(this, obc),
 					    new C_OSD_CommittedPushedObject(
 					      this,
 					      get_osdmap()->get_epoch(),
 					      info.last_complete),
 					    new C_OSD_OndiskWriteUnlock(obc));
+              delete t;
```

Comment:
we should be able to avoid using new/delete entirely with this new interface. so `t` here should be on the stack, instead of heap-allocated above